packages/opencensus-instrumentation-mongodb/src/mongodb.ts

```diff
@@ -0,0 +1,162 @@
+ * limitations under the License.
+ */
+
+import {types} from '@opencensus/opencensus-core';
+import {classes} from '@opencensus/opencensus-core';
+import {logger} from '@opencensus/opencensus-core';
+import * as mongodb from 'mongodb';
+import * as semver from 'semver';
+import * as shimmer from 'shimmer';
+import {callbackify} from 'util';
+
+
+/** Http instrumentation plugin for Opencensus */
+export class MongoDBPlugin extends classes.BasePlugin {
+  readonly SERVER_FNS = ['insert', 'update', 'remove', 'auth'];
+  readonly CURSOR_FNS_FIRST = ['_find', '_getmore'];
+  readonly SPAN_MONGODB_QUERY_TYPE = 'db.mongodb.query';
+  logger: types.Logger;
+
+  /** Constructs a new HttpPlugin instance. */
+  constructor(moduleName: string) {
+    super(moduleName);
+  }
+
+  // tslint:disable-next-line:no-any
+  applyPatch(moduleExports: any, tracer: types.Tracer, version: string) {
+    this.setPluginContext(moduleExports, tracer, version);
+    this.logger = tracer.logger || logger.logger('debug');
+
+    this.logger.debug('Patched MongoDB');
+
+    if (moduleExports.Server) {
+      this.logger.debug('patching mongodb-core.Server.prototype.command');
+      shimmer.wrap(
+          moduleExports.Server.prototype, 'command', this.patchCommand());
+      this.logger.debug(
+          'patching mongodb-core.Server.prototype functions:', this.SERVER_FNS);
+      shimmer.massWrap(
+          [moduleExports.Server.prototype], this.SERVER_FNS, this.patchQuery());
+    }
+
+    if (moduleExports.Cursor) {
+      this.logger.debug(
+          'patching mongodb-core.Cursor.prototype functions:',
+          this.CURSOR_FNS_FIRST);
+      shimmer.massWrap(
+          [moduleExports.Cursor.prototype], this.CURSOR_FNS_FIRST,
+          this.patchCursor());
+    }
+
+    return moduleExports;
+  }
+
+  /** Unpatches all HTTP patched function. */
+  applyUnpatch(): void {
+    shimmer.unwrap(this.moduleExports.Server.prototype, 'command');
+    shimmer.massUnwrap(this.moduleExports.Server.prototype, this.SERVER_FNS);
+    shimmer.massUnwrap(
+        this.moduleExports.Cursor.prototype, this.CURSOR_FNS_FIRST);
+  }
+
+  patchCommand() {
+    const plugin = this;
+    return (original: types.Func<mongodb.Server>) => {
+      // tslint:disable-next-line:no-any
+      return function(this: mongodb.Server, ns: string, command: any):
+          mongodb.Server {
+            let span: types.Span;
+            const index = arguments.length - 1;
+            const resultHandler = arguments[index];
+
+            if (plugin.tracer.currentRootSpan && arguments.length > 0 &&
+                typeof resultHandler === 'function') {
+              let type: string;
+              if (command.createIndexes) {
+                type = 'createIndexes';
+              } else if (command.findandmodify) {
+                type = 'findAndModify';
+              } else if (command.ismaster) {
+                type = 'ismaster';
```

Comment:
Make this camelcase for consistency (or make others not camelcase)

Reply:
Done.